packages/components/src/components/DistributionChart.tsx

```diff
@@ -83,18 +83,26 @@ export const DistributionChart: React.FC<DistributionChartProps> = (props) => {
       }
     }
 
-    const spec = buildVegaSpec(props);
+    const domain = shapes.value.flatMap((shape) =>
+      shape.discrete.concat(shape.continuous)
+    );
+
+    const spec = buildVegaSpec({
+      ...props,
+      minX: props.minX ?? Math.min(...domain.map((x) => x.x)),
+      maxX: props.minX ?? Math.max(...domain.map((x) => x.x)),
+      maxY: Math.max(...domain.map((x) => x.y)),
+    });
 
-    let widthProp = width ? width : size.width;
+    let widthProp = width ? width : isFinite(size.width) ? size.width : 400;
```

Comment:
My guess is that this fixes the docusaurus problem with rendering in tabs and so on? Might be worthwhile to leave a comment on that.

Reply:
I thought that too! I'm not really sure though. I'll add a comment as to why size.width may not a finite number.